src/utils/config.py

```diff
@@ -0,0 +1,216 @@
+"""Simple utils to assist the config."""
+from copy import deepcopy
+from typing import Union
+
+from src.utils.defence import (
+    type_defence, 
+    validate_file_extension
+)
+from src.utils.local_file_mods import safeload_yaml
```

Comment:
What if we're running on hdfs ? (funnily enough, it doesn't break on devtest)

Reply:
The config is always expected to be read in locally here,  not from HDFS.

Reply:
OK, so even when you're on DAP, you're using unix rather than windows, but you're not accessing the hdfs system as you're just reading another file in the package (which presumably is not on the hdfs system.) Is that right?